xtraplatform-feature-provider-sql/src/main/java/de/ii/xtraplatform/feature/provider/sql/infra/db/SqlTypeInfoValidator.java

```diff
@@ -103,13 +142,31 @@ public SqlTypeInfoValidator(ConnectionInfoSql connectionInfo, ClassLoader classL
     }
 
     private boolean tableExists(String name) {
-        //TODO: check this.tables
-        return false;
+        return tables.stream().anyMatch(t -> t.getName().equals(name));
     }
 
     private boolean columnExists(String name, String table) {
-        //TODO: check this.tables
-        return false;
+        return tables.stream()
+                .filter(t -> t.getName().equals(table))
+                .flatMap(t -> t.getColumns().stream())
+                .anyMatch(c -> c.getName().equals(name));
+    }
+
+    private boolean isColumnUnique(String name, String table) {
+        Optional<Column> column = tables.stream()
+                .filter(t -> t.getName().equals(table))
+                .flatMap(t -> t.getColumns().stream())
+                .filter(c -> c.getName().equals(name))
+                .findFirst();
+        return column.isPresent() && (column.get().isPartOfPrimaryKey() || column.get().isPartOfUniqueIndex());
```

Comment:
I think `isPartOf` might not be strict enough, the index should contain only this column.

For primary key you can check that on the table: 

```
table.getPrimaryKey().getColumns().size() == 1 && table.getPrimaryKey().getColumns().get(0).getReferencedColumn().equals(column)
```

For unique indexes you would have to iterate over `table.getIndexes()`, filter by `isUnique()` and then check like above that `getColumns()` only contains the given `column`.